src/slice_file.rs

```diff
@@ -168,6 +169,67 @@ impl SliceFile {
     }
 }
 
+pub trait SliceFileHashable {
+    /// Hashes the SliceFile using a SHA-256 hash and returns the 32-byte array.
+    fn compute_sha256_hash_as_bytes(&self) -> [u8; 32];
+
+    /// Hashes the SliceFile using a SHA-256 hash and returns the hash as a hex string.
+    fn compute_sha256_hash(&self) -> String;
+}
+
+impl SliceFileHashable for SliceFile {
+    /// Hash the combination of the filename and the raw text using a SHA-256 hash.
+    ///
+    /// # Returns
+    /// The SHA-256 hash as a 32-byte array.
+    fn compute_sha256_hash_as_bytes(&self) -> [u8; 32] {
+        Sha256::new()
+            .chain_update(self.filename.as_bytes())
+            .chain_update(self.raw_text.as_bytes())
+            .finalize()
+            .into()
+    }
+
+    /// Hash the combination of the filename and the raw text using a SHA-256 hash.
+    ///
+    /// # Returns
+    /// The SHA-256 hash as a hex string.
+    fn compute_sha256_hash(&self) -> String {
+        self.compute_sha256_hash_as_bytes()
+            .iter()
+            .map(|byte| format!("{:02x}", byte))
+            .collect()
+    }
```

Comment:
And then you can delete these duplicates:
```suggestion
    }
```